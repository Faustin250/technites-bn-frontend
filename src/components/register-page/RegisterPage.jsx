/* eslint-disable arrow-parens */
/* eslint-disable comma-dangle */
/* eslint-disable jsx-a11y/label-has-associated-control */
/* eslint-disable react/button-has-type */
/* eslint-disable quotes */
import React from "react";
import { connect } from "react-redux";
import Joi from "joi-browser";
import Form from "../form/form";
import { register } from "../../redux/actions/registerAction";
import NavBar from "../navbar/navbar";

export class Register extends Form {
  doSubmit = async () => {
    this.props.register(this.state.data);
  };

  schema = {
    firstname: Joi.string()
      .required()
      .min(3)
      .label("First Name"),
    lastname: Joi.string()
      .required()
      .min(3)
      .label("Last Name"),
    username: Joi.string()
      .required()
      .min(3)
      .label("Username"),
    email: Joi.string()
      .email()
      .required()
      .label("Email"),
    password: Joi.string()
      .required()
      .min(6)
      .label("Password"),
    confirmPassword: Joi.string()
      .required()
      .min(6)
      .label("Confirm Password")
  };

  render() {
    return (
      <div className="container">
        <NavBar />
        <div className="card-container">
          <div className="card">
            <div className="title">Create an account</div>
            {this.renderInput("firstname", "First Name")}
            {this.renderInput("lastname", "Last Name")}
            {this.renderInput("username", "Username")}
            {this.renderInput("email", "Email")}
            {this.renderInput("password", "Password", "password")}
            {this.renderInput(
              "confirmPassword",
              "Confirm Password",
              "password"
            )}
            {this.renderButton("Sign up", "register")}
          </div>
        </div>
      </div>
    );
  }
}

export const mapStateToProps = ({ register }) => ({
  users: register
});

export default connect(mapStateToProps, { register })(Register);
