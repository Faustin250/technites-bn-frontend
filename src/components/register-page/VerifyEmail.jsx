import React, { Component } from "react";
import NavBar from "../navbar/navbar";
import "../form/form.scss";
import image from "../../assets/verify.png";

class VerifyEmailPage extends Component {
  render() {
    return (
      <div>
        <NavBar />
        <div className="card-container">
          <form className="card" onSubmit={this.handleSubmit}>
            <div className="title">Verify Email</div>
            <img className="margin" src={image} alt="" />
            <p className="text">
              You are almost ready to start enjoying Barefoot Nomad, Just verify
              your email using the link we sent you!
            </p>
          </form>
        </div>
      </div>
    );
  }
}

export default VerifyEmailPage;
