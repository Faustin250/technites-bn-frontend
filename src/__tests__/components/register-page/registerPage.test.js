/* eslint-disable import/no-named-as-default-member */
/* eslint-disable quotes */
/* eslint-disable import/named */
/* eslint-disable no-undef */
import React from "react";
import { mount } from "enzyme";
import { Register, mapStateToProps as mapStateToPropsNew } from '../../../components/register-page/RegisterPage';

describe("<RegisterPage />", () => {
  const wrapper = mount(<Register />);
  const wrapper2 = mount(<Register />);
  const props = {
    register: jest.fn(),
    displayVerifyConfirmation: false,
    auth: false
  }
  wrapper.setProps(props);
  props.displayVerifyConfirmation = true;
  wrapper2.setProps(props);

  it("should taste the errors functionality", () => {
    const event = { target: { name: 'firstname', value: 'am' } };
    const input = wrapper.find('input[name="firstname"]');
    input.simulate('change', event);
    const { errors } = wrapper.state();
    expect(errors[event.target.name]).toContain('length must be at least');
  });

  it("should check if data are valid when you send the data without typing", () => {
    wrapper.find('button[className="button"]');
  });

  it("should check if password and confirm password are the same", () => {
    const values = [
      { target: { name: 'firstname', value: 'test' } },
      { target: { name: 'lastname', value: 'test' } },
      { target: { name: 'username', value: 1 } },
      { target: { name: 'email', value: 'test@gmail.com' } },
      { target: { name: 'password', value: 'test1#' } },
      { target: { name: 'confirmPassword', value: 'test1@' } }];

    const input = wrapper.find('input[name="firstname"]');
    values.forEach((value) => {
      input.simulate('change', value);
    });

    // const form = wrapper.find('form[className="card"]');
    const button = wrapper.find('button[className="button"]');

    button.simulate('click');

    // const { data } = wrapper.state();
    // expect(data[values[0].target.name]).toEqual(values[0].target.value);
  });

  it("should check if the input field are being rendered", () => {
    const values = [
      { target: { name: 'firstname', value: 'test' } },
      { target: { name: 'lastname', value: 'test' } },
      { target: { name: 'username', value: 'test' } },
      { target: { name: 'email', value: 'test@gmail.com' } },
      { target: { name: 'password', value: 'test1#' } },
      { target: { name: 'confirmPassword', value: 'test1#' } }];
    // const event = { target: { name: 'firstname', value: 'test' } };

    const input = wrapper.find('input[name="firstname"]');
    values.forEach((value) => {
      input.simulate('change', value);
    });

    const { data } = wrapper.state();
    expect(data[values[0].target.name]).toEqual(values[0].target.value);
  });

  it("should test for register a user", () => {
    const button = wrapper.find('button[className="button"]');
    button.simulate('click');
  });
  it("should test view after successful submit", () => {
    const componentSuccess = wrapper2.find('[data-test="success-view"]');
    expect(componentSuccess.length).toBe(1);
  });

  it("should test map state to props of register", () => {
    mapStateToPropsNew({ register: {}, loginState: {} });
  });
});
