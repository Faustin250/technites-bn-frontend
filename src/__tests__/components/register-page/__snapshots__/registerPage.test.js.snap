// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RegisterPage /> should render the register component 1`] = `ShallowWrapper {}`;
